### ISSUE TITLE
Investigate possibility of generating checksums client-side to validate upload has been bit-perfect

### ISSUE THREAD

**Issue by jesusbagpuss** [eprints/eprints3.4#259]
NB This is a 'for discussion' issue - rather than something currently broken.

Would it be desirable to be able to generate a checksum for a file prior to upload, and then validate it against the checksum of the uploaded file?

Currently I don't think there is any validation at the point a file is received.

There are various javascript libraries that could be used, many discussed on the post that this comment belongs to: 
https://stackoverflow.com/a/63287199
The solution proposed in the above answer uses https://www.npmjs.com/package/hash-wasm - supported by all(?) modern browsers.

**Comment by drn05r** [comment]
@dgc has been looking into improving the JS libraries and associate code around file uploads to deal with various occasional upload issues and to make upload more efficient.  This does not specifically include using hashes to validate file uploads but may be worth considering in the changes he has been working on.

**Comment by dgc** [comment]
I agree with this idea and also the proposed package to do the hashing.

Modern browsers have hashing already via the Crypto API but that requires the file content to be in memory all at once.
